lib/ht_sip_validator/validator/ocr.rb

```diff
@@ -2,6 +2,26 @@
 
 # namespace for validators for meta.yml
 module HathiTrust::Validator::OCR
+  # Returns an error for each file in the set 'other' that isn't in the set 'base'
+  # using the given names of the sets and file extension for the 'other' set
+  def file_set_diff(base, base_name, other, other_name, other_ext, message_level)
+    message_level_method = "create_#{message_level}"
+
+    base.keys.to_set.difference(other.keys.to_set).map do |seq|
+      send(message_level_method,
```

Comment:
At the very least prefer public_send, but this might be a sign that an abstraction is missing.